sktime/transformations/series/temporian.py

```diff
@@ -0,0 +1,115 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Implementation of TemporianTransformer. Contributed by the Temporian team."""
+
+__author__ = ["ianspektor"]  # TODO: add other Temporian devs who contributed to this
+
+from sktime.transformations.base import BaseTransformer
+
+
+# TODO: add usage example to docstring
+# see https://www.sktime.net/en/latest/developer_guide/dependencies.html#dependencies
+# for how to handle soft deps (temporian) there (need to not run doc tests on it)
+class TemporianTransformer(BaseTransformer):
+    """Applies a Temporian function to the input time series.
+
+    This transformer applies a [Temporian](https://temporian.readthedocs.io/en/stable/)
+    function to the input time series.
+
+    The conversion from sktime's internal representation to Temporian's
+    [EventSet](https://temporian.readthedocs.io/en/stable/reference/temporian/EventSet/)
+    and back is handled automatically by the transformer.
+
+    Parameters
+    ----------
+    function : Callable[[temporian.EventSet], temporian.EventSet]
+        Temporian function to apply to the input time series. The function must receive
+        and return a single Temporian EventSet, and can apply an arbitrary number of
+        Temporian operators to its input.
+    """
+
+    _tags = {
+        "univariate-only": False,
+        "authors": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who contributed to this
+        "maintainers": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who will maintain this
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        # TODO: extend to pd-multiindex and pd_multiindex_hier
+        "X_inner_mtype": "pd.DataFrame",
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "python_dependencies": ["temporian"],
+        "python_version": "<3.12",
```

Comment:
does `temporian` not support python 3.12? I.e., is the version bound intended or a typo?

Reply:
it's intended, we don't support 3.12 atm, see https://github.com/google/temporian/blob/main/pyproject.toml#L36

Reply:
I'll suggest to keep only a lower bound here. Even if you don't add a upper bound, temporian will still be not installed in a 3.12 environment since it's not available on PyPI. Then soft dependency check will prevent use of this transformer already and this check will not be used.

But if you keep this and later starts supporting 3.12, it's easy to miss here as probably we/dependabot will see only pyproject.toml and users will not be able to use this on 3.12.

This is of course just a suggestion, please let me know what you think.

Reply:
Sounds good 👍🏼 cc. @javiber